code/campaignclassic/src/main/java/com/adobe/marketing/mobile/AEPMessagingService.java

```diff
@@ -105,9 +115,15 @@ public static boolean handleRemoteMessageData(
         AEPPushPayload payload;
         try {
             payload = new AEPPushPayload(messageData);
+            // get the tag from the payload. if no tag was present in the payload use the message id
+            // instead as its guaranteed to always be present.
+            final String tag =
+                    !StringUtils.isNullOrEmpty(payload.getTag())
+                            ? payload.getTag()
+                            : payload.getMessageId();
```

Comment:
Why not use the `messageId` always if it is guaranteed to be present?

Reply:
the tag is used in the case where the marketer would like a new notification to replace an existing one in the notification drawer. the message id is different for each message sent from ACC but having the same tag would allow a different message sent from FCM to replace an existing one.

Reply:
Got it. So is my understanding that _the tag optional but when present it supersedes the message id_  correct?

Reply:
that is correct